maas/data_source_maas_boot_source.go

```diff
@@ -0,0 +1,68 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceMaasBootSource() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMaasBootSourceRead,
+		Description: "Provides a resource to fetch MAAS boot sources.",
```

Comment:
I think after the change to fetch the one and only boot source, it makes sense to update the description like this. Note that this change will require re-generating the docs

```suggestion
		Description: "Provides a resource to fetch the MAAS boot source.",
```